lib/streamlit/elements/toast.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Optional, cast
+
+from streamlit.errors import StreamlitAPIException
+from streamlit.proto.Toast_pb2 import Toast as ToastProto
+from streamlit.runtime.metrics_util import gather_metrics
+from streamlit.string_util import clean_text, validate_emoji
+from streamlit.type_util import SupportsStr
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+
+
+def validate_type(toast_type: Optional[str]) -> str:
+    valid_types = ["success", "warning", "error"]
+
+    if toast_type is None:
+        return ""
+
+    toast_type = toast_type.lower()
+    if toast_type in valid_types:
+        return toast_type
+    else:
+        raise StreamlitAPIException(
+            f"Invalid toast type: {toast_type}. Valid types are “success”, “warning”, “error”, or None"
+        )
+
+
+def validate_text(toast_text: SupportsStr) -> None:
+    if toast_text is "":
+        raise StreamlitAPIException(
+            f"Toast text cannot be blank - please provide a message."
+        )
+
+
+class ToastMixin:
+    @gather_metrics("toast")
+    def toast(
+        self,
+        text: SupportsStr,
+        *,  # keyword-only args:
+        icon: Optional[str] = None,
+        type: Optional[str] = None,
+    ) -> "DeltaGenerator":
+        """Display a toast in the bottom right corner of the screen. Will disappear after four seconds.
+        Parameters
+        ----------
+        text : str
+            Short message for the toast.
+        icon : str or None
+            An optional, keyword-only argument that specifies an emoji to use as
+            the icon for the toast. Shortcodes are not allowed, please use a
+            single character instead. E.g. "🚨", "🔥", "🤖", etc.
+            Defaults to None, which means no icon is displayed.
+        type : “success”, “warning”, “error”, or None
+            An optional, keyword-only argument that specifies the type of toast.
+            Defaults to None.
```

Comment:
Thanks for flagging, @mayagbarnes! The original version is straightforward and easy to understand, but could use a bit more detail about what a "toast" is. I have a hunch it isn't obvious to many. WDYT about the revision below? Note: the product spec has some recent comments about temporarily omitting the `type` kwarg for the initial release, but I've still slightly improved its description for the time being.

Feel free to update the docstring for `text` wrt to its Markdown support, based on the latest comments in the spec.

The API reference docs page will also include embedded examples (which I can share in the coming days) containing all the toast types (or just the default one if `type` is temporarily omitted).

```suggestion
        """Display a short message, known as a notification "toast".

        The toast appears in the app's bottom-right corner and disappears after four seconds.

        Parameters
        ----------
        text : str
            Short message for the toast.
        icon : str or None
            An optional, keyword-only argument that specifies an emoji to use as
            the icon for the toast. Shortcodes are not allowed, please use a
            single character instead. E.g. "🚨", "🔥", "🤖", etc.
            Defaults to None, which means no icon is displayed.
        type : "success", "warning", "error", or None
            An optional, keyword-only argument that specifies the type of toast, which
            influences its color. Defaults to None, which corresponds to a neutral gray.
```

Reply:
I had to Google "notification toast", so, yes agree users need explanation.  Frankly, I am not crazy about "toast" as the name for a short popup.  I feel like only UX designers know this term.  What about st.short_popup?